apps/api/src/app/shared/services/encryption/cipher.ts

```diff
@@ -0,0 +1,26 @@
+import { createCipheriv, createDecipheriv, randomBytes } from 'crypto';
+
+const ENCRYPTION_KEY = process.env.P_ENCRYPTION_KEY;
+const IV_LENGTH = 16;
+
+export function encrypt(text) {
+  const iv = randomBytes(IV_LENGTH);
+  const cipher = createCipheriv('aes-256-cbc', Buffer.from(ENCRYPTION_KEY), iv);
+  let encrypted = cipher.update(text);
+
+  encrypted = Buffer.concat([encrypted, cipher.final()]);
+
+  return iv.toString('hex') + ':' + encrypted.toString('hex');
+}
+
+export function decrypt(text) {
+  const textParts = text.split(':');
+  const iv = Buffer.from(textParts.shift(), 'hex');
+  const encryptedText = Buffer.from(textParts.join(':'), 'hex');
+  const decipher = createDecipheriv('aes-256-cbc', Buffer.from(ENCRYPTION_KEY), iv);
+  let decrypted = decipher.update(encryptedText);
+
+  decrypted = Buffer.concat([decrypted, decipher.final()]);
+
+  return decrypted.toString();
+}
```

Comment:
Would be great to have unit tests and typescript annotations here aswell 🙏

Reply:
Agree i really wanted to add a unit test here but missed the rush of continuing, however, I did add an e2e test on the new usecase I created on the next PR but the unit test will be much more appropriate here I will look into it.! :)